### PR TITLE
Require program slugs to be alphanumeric

### LINKED ISSUE #6163: Require program slugs to be alphanumeric
In [Design: Streamlining URL Schema for Applicants](https://docs.google.com/document/d/1bx95UApQAJ9tJnnOSkxgn4jx3ewHE5z3uCZtb-Rl8A4/edit), we note a route collision between programs being fetched by ID vs programs being fetched by program slug. The usage is disambiguated by checking to see if the path component is purely numeric; if so, it is considered a program id.

Enforce a requirement that program slugs be alphanumeric in order to enable this disambiguation.

Relates to #5576

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Make sure admin name contains at least one letter, otherwise it  is considered a program id.

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

#### User visible changes

- [ ] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
- [ ] Wrote browser tests using the [validateAccessibility](https://sourcegraph.com/github.com/civiform/civiform/-/blob/browser-test/src/support/index.ts?L437:14&subtree=true) method
- [ ] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [ ] Manually tested at 200% size
- [ ] Manually evaluated tab order

### Issue(s) this completes

Fixes #6163

### CODE REVIEW

**Review by yotommy — APPROVED** [review]
Thank you!

File: server/test/services/program/ProgramServiceTest.java [hunk 1]
```diff
@@ -459,6 +459,22 @@ public void validateProgramDataForCreate_requiresSlug(String adminName) {
...
+            "display name",
+            "display desc",
+            "https://usa.gov",
+            DisplayMode.PUBLIC.getValue(),
+            ImmutableList.copyOf(new ArrayList<>()));
+
+    assertThat(result)
+        .containsExactly(CiviFormError.of("A program URL must contain at least one letter"));
+  }
+
```
Comment (avaleske) [inline server/test/services/program/ProgramServiceTest.java:477]: Can we also add a test that tests that it passes with one letter?

We can append something like `_failsIfNoLetters` and `_validIfOneLetter` to the end of the test names to disambiguate.

File: server/app/services/program/ProgramService.java [hunk 2]
```diff
@@ -69,6 +70,8 @@ public final class ProgramService {
   private static final String MISSING_ADMIN_NAME_MSG = "A program URL is required";
   private static final String INVALID_ADMIN_NAME_MSG =
       "A program URL may only contain lowercase letters, numbers, and dashes";
+  private static final String INVALID_PROGRAM_SLUG_MSG =
+      "A program URL must contain at least one letter";
```
Comment (avaleske) [inline server/app/services/program/ProgramService.java:74]: Optional: What does the form text use? Does it call the field a URL?

Would something like "A program identifier must contain..." better align with the form?

**Review by avaleske — CHANGES_REQUESTED** [review]
Thank you! Just need a test for the passing condition

File: server/app/services/program/ProgramService.java (hunk 2 above)
Comment (azhangggggg) [reply server/app/services/program/ProgramService.java:74]: This error message is referring to the same input box as the ones on line 72 `INVALID ADMIN_NAME_MSG` and `MISSING_ADMIN_NAME_MSG` so I chose the same subject in the error message to align with them. 

Can change it if you feel strongly about changing the subject tho!

File: server/test/services/program/ProgramServiceTest.java (hunk 1 above)
Comment (azhangggggg) [reply server/test/services/program/ProgramServiceTest.java:478]: for sure, a test with it passes with one letter added!

File: server/app/services/program/ProgramService.java (hunk 2 above)
Comment (avaleske) [reply server/app/services/program/ProgramService.java:74]: Ah makes sense, that works for me.

**Review by avaleske — APPROVED** [review]

**Pull request merged**